Automattic/wp-calypso
# Adds & uses hoverintent component

Build on https://github.com/Automattic/wp-calypso/pull/47238
#### Changes proposed in this Pull Request

* Added https://www.npmjs.com/package/react-hoverintent npm package and used it in `client/layout/sidebar/expandable.jsx` 
* `sensitivity`, `interval={ 90 }`, `timeout={ 200 }` are the same as in wp-admin
* This change will not affect production unless you use `?flags=nav-unification` in URL

#### Testing instructions



* add `?flags=nav-unification` in URL
* hover fast the sidebar menus vertically, none should open
* hover fast the sidebar menus vertically and stay over one randomly, the flyout should open.
* hover the Setting menu, try moving the cursor to General in a straight line (fast enough). The flyout should not close.
* hover the Setting menu, try moving the cursor to General in a straight line (slowly). The flyout should close.
* hover the Posts menu, try moving the cursor to Tags in a straight line(fast enough). The flyout should not close. Links flyout should not open.

Before | After
-------|------
![SS 2020-11-10 at 13 48 18](https://user-images.githubusercontent.com/12430020/98670528-7d881b00-235b-11eb-9030-8ac1c36bbad3.gif)|![SS 2020-11-10 at 13 49 20](https://user-images.githubusercontent.com/12430020/98670627-a3152480-235b-11eb-82e9-809c626f538f.gif)

Fixes https://github.com/Automattic/wp-calypso/issues/46094
